internal/registry/registry.go

```diff
@@ -16,38 +17,40 @@ import (
 // imports and ensures there are no conflicts in the imported package
 // qualifiers.
 type Registry struct {
-	srcPkg     *packages.Package
-	moqPkgPath string
-	aliases    map[string]string
-	imports    map[string]*Package
+	srcPkgName  string
+	srcPkgTypes *types.Package
+	moqPkgPath  string
+	aliases     map[string]string
+	imports     map[string]*Package
 }
 
 // New loads the source package info and returns a new instance of
 // Registry.
 func New(srcDir, moqPkg string) (*Registry, error) {
 	srcPkg, err := pkgInfoFromPath(
-		srcDir, packages.NeedName|packages.NeedSyntax|packages.NeedTypes|packages.NeedTypesInfo|packages.NeedDeps,
+		srcDir, packages.NeedName|packages.NeedSyntax|packages.NeedTypes,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't load source package: %s", err)
 	}
 
 	return &Registry{
-		srcPkg:     srcPkg,
-		moqPkgPath: findPkgPath(moqPkg, srcPkg),
-		aliases:    parseImportsAliases(srcPkg),
-		imports:    make(map[string]*Package),
+		srcPkgName:  srcPkg.Name,
+		srcPkgTypes: srcPkg.Types,
+		moqPkgPath:  findPkgPath(moqPkg, srcPkg.PkgPath),
+		aliases:     parseImportsAliases(srcPkg.Syntax),
+		imports:     make(map[string]*Package),
```

Comment:
`srcPkg` only has the fields populated that were instructed to be populated by the `packages.LoadMode` (line 31). It is therefore not ideal to let `srcPkg` outside of this `New` function because it opens up the possibility for other areas of the code to depend on fields that were not loaded. It is also more difficult to see if other areas of the code actually require certain `packages.LoadMode` flags. By not passing `srcPkg` outside of this `New` function, it is now easier to see which `packages.LoadMode` flags are truly needed.